griptape/structures/agent.py

```diff
@@ -38,18 +39,20 @@ def validate_fail_fast(self, _: Attribute, fail_fast: bool) -> None:  # noqa: FB
     def __attrs_post_init__(self) -> None:
         super().__attrs_post_init__()
 
-        self.prompt_driver.stream = self.stream
+        # Make a copy to avoid unintentionally modifying the global default Prompt Driver's stream setting
+        prompt_driver = copy(self.prompt_driver)
+        prompt_driver.stream = self.stream
```

Comment:
we aren't setting `self.prompt_driver` with the copy?

Reply:
`self.prompt_driver` only exists as a convenient way to set the prompt drivers on the Tasks. It doesn't get used anywhere else besides here so I don't think it makes sense to overwrite it with a new value.

Reply:
but i could still modify the global default by doing `agent.prompt_driver.stream = True`? and then the behavior of my agent wouldnt reflect that setting change

Reply:
i liked this: `Factory(lambda: copy(Defaults.drivers_config.prompt_driver))`

Reply:
omg: https://www.attrs.org/en/stable/api.html#attrs.evolve